docs/reference/env-variables.rst

```diff
@@ -0,0 +1,688 @@
+    *
+      - AMD_LOG_LEVEL_FILE
+      - cstring
+      - 
+      - Set output file for AMD_LOG_LEVEL. Default is stderr
+    *
+      - PAL_PREPINNED_MEMORY_SIZE
+      - size_t
+      - 64
+      - Size in KBytes of prepinned memory
+    *
+      - AMD_CPU_AFFINITY
+      - bool
+      - FALSE
+      - Reset CPU affinity of any runtime threads
+    *
+      - ROC_USE_FGS_KERNARG
+      - bool
+      - TRUE
+      - Use fine grain kernel args segment for supported ASICs
+    *
+      - ROC_P2P_SDMA_SIZE
+      - uint
+      - 1024
+      - The minimum size in KB for P2P transfer with SDMA
+    *
+      - ROC_AQL_QUEUE_SIZE
+      - uint
+      - 16384
+      - AQL queue size in AQL packets
+    *
+      - ROC_SIGNAL_POOL_SIZE
+      - uint
+      - 32
+      - Initial size of HSA signal pool
+    *
+      - DEBUG_CLR_LIMIT_BLIT_WG
+      - uint
+      - 16
+      - Limit the number of workgroups in blit operations
+    *
+      - ROC_SKIP_KERNEL_ARG_COPY
+      - bool
+      - FALSE
+      - If true, then runtime can skip kernel arg copy
+    *
+      - GPU_STREAMOPS_CP_WAIT
+      - bool
+      - FALSE
+      - Force the stream wait memory operation to wait on CP.
+    *
+      - HIP_USE_RUNTIME_UNBUNDLER
+      - bool
+      - FALSE
+      - Force this to use Runtime code object unbundler.
+    *
+      - HIPRTC_USE_RUNTIME_UNBUNDLER
+      - bool
+      - FALSE
+      - Set this to true to force runtime unbundler in hiprtc.
+    *
+      - HIP_INITIAL_DM_SIZE
+      - size_t
+      - 8388608
+      - Set initial heap size for device malloc. The default value corresponds to 8 MiB
+    *
+      - HIP_FORCE_DEV_KERNARG
+      - bool
+      - 0
+      - Force device mem for kernel args.
+    *
+      - DEBUG_CLR_GRAPH_PACKET_CAPTURE
+      - bool
+      - FALSE
+      - Enable/Disable graph packet capturing
+    *
+      - GPU_DEBUG_ENABLE
+      - bool
+      - FALSE
+      - Enables collection of extra info for debugger at some performance cost
+    *
+      - HIPRTC_COMPILE_OPTIONS_APPEND
+      - cstring
+      - 
+      - Set compile options needed for hiprtc compilation
+    *
+      - HIPRTC_LINK_OPTIONS_APPEND
+      - cstring
+      - 
+      - Set link options needed for hiprtc compilation
+    *
+      - HIP_VMEM_MANAGE_SUPPORT
+      - bool
+      - TRUE
+      - Virtual Memory Management Support
+    *
+      - HIPCC_VERBOSE
+      - uint
+      - 0
+      - How much extra info to show during build. E.g: compiler flags, paths.
+    *
+      - ROCPROFILER_REGISTER_ROOT
+      - cstring
+      - 
+      - The path to the rocProfiler.
+
+AMD rocclr environment variables at debug build:
+
+.. list-table:: AMD rocclr environment variables
+    :header-rows: 1
+    :name: rocclr-env-variables-debug-table
+
+    *
+      - Environment variable
+      - Variable type
+      - Default value
+      - Description
+    *
+      - DEBUG_GPU_FLAGS
```

Comment:
We should not expose internal variable for runtime debug.

Reply:
Basically don't copy all variables from flags.hpp into this doc...

Reply:
Should we have different sections about the variables? Like common ones, hacker ones, debug ones?
If they do exist, it would be nice they are described somewhere from a software quality perspective.

Reply:
ok, i can check and remove those. there are some HSA env variables, that also look like they are for internal debugging, should i remove those as well? (e.g: HSA_OVERRIDE_GFX_VERSION)

Reply:
At the end `HSA_OVERRIDE_GFX_VERSION` seems quite necessary since it allows some owners of non-supported GPU to just use them.